cpp/src/arrow/dataset/dataset.h

```diff
@@ -44,6 +44,8 @@ class ARROW_DS_EXPORT DataFragment {
   /// scanning
   virtual bool splittable() const = 0;
 
+  virtual std::string type() const = 0;
```

Comment:
Call this `type_name()`.

Reply:
Well, `type()` and `name()` are already used across the codebase. I'd favor to introduce a naming convention or either mixins to help the bindings a bit.

Reply:
Certainly not `type()`, which is for a type id or object. `name()` is for a per-instance name. So `type_name()` is the right naming IMHO.

Reply:
Modified them.